elasticsearch/src/main/java/org/hibernate/search/backend/elasticsearch/impl/ElasticSearchHSQueryImpl.java

```diff
@@ -100,11 +115,8 @@ public DocumentExtractor queryDocumentExtractor() {
 
 	@Override
 	public int queryResultSize() {
-		if ( resultSize == null ) {
```

Comment:
This change introduces a bug as searchResult is not reset in clearCachedResults(). I fixed it in my branch by also resetting it but if you plan to merge this sooner rather than later, I thought it was worth mentioning.